packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -114,7 +116,9 @@ const resolveFixed = (image, options) => {
   fixedSizes = fixedSizes.map(Math.round)
 
   // Filter out sizes larger than the image's width.
-  const filteredSizes = fixedSizes.filter(size => size <= width)
+  const filteredSizes = fixedSizes.filter(
+    size => size <= contentful_max_size || size <= width
```

Comment:
I've been testing this condition and I think you want `&&` as we want to filter out images that are bigger than 4000 or bigger than the image width. By changing this the test will fail but that's a good thing :wink: 

```suggestion
    size => size <= contentful_max_size && size <= width
```

Reply:
Oops! What a mistake!